source/magpie/dataframe.d

```diff
@@ -1098,6 +1098,68 @@ public:
                 data[i][pos] = elements.data[i];
         }
     }
+
+    /++
+    void apply(alias Fn, int axis, T)(T index)
+    Description: Applies a function on a particular row or column
+    @params: Fn - Function
+    @params: axis - 0 for rows, 1 for columns
+    @params: index - integer or string indexes of rows
+    +/
+    void apply(alias Fn, int axis, T)(T index)
+        if(is(T == int[]) || is(T == string[][]))
```

Comment:
You probably want to add a constraint to `Fn` to make sure that it takes and returns a `typeof(data[0][0])`

Reply:
For now I have kept the implementation very forgiving.
I'll probably add a `__compile` soon to check if `Fn(data[i])` is possible.